src/react/DevicesPage.jsx

```diff
@@ -35,126 +36,150 @@ import SingleCardPage from './ui/SingleCardPage';
 import { useAlerts } from './AlertManager';
 
 const DEVICES_PER_PAGE = 20;
+const DEVICES_PER_REQUEST = 100;
 const MAX_SHOWN_PAGES = 10;
 
+const matchFilters = (device, filters) => {
+  const {
+    deviceId = '',
+    showConnected = true,
+    showDisconnected = true,
+    showNeverConnected = true,
+  } = filters;
+
+  if (!showConnected && device.connected) {
+    return false;
+  }
+  if (!showDisconnected && !device.connected && device.lastConnection) {
+    return false;
+  }
+  if (!showNeverConnected && !device.connected && !device.lastConnection) {
+    return false;
+  }
+  if (deviceId === '') {
+    return true;
+  }
+
+  const aliases = Array.from(device.aliases.values());
+  return (
+    device.id.includes(deviceId) || aliases.filter((alias) => alias.includes(deviceId)).length > 0
```

Comment:
You can change `aliases.filter((alias) => alias.includes(deviceId)).length > 0` to `aliases.some((alias) => alias.includes(deviceId))` so it is easier to read